### DISCUSSION TITLE
Change persistence layer

### DISCUSSION THREAD

**Discussion by jsmassa** [replikativ/datahike#491] [Ideas]
Since we want to switch to the persistent set as main index structure, we have to decide if we want to keep doing it the way it is implemented at the moment for the hitchhiker tree or if there is a better way to save the data to with the chosen backend. I wrote an [ADR](https://github.com/replikativ/datahike/blob/persisten-set-adr/doc/adl/adr-005-persistence.md) for that but there might be other options I haven't thought of yet. 

We could also discuss different implementations depending on the user's performance needs.

**Comment by whilo** [comment]
I have pushed the explorative work I have done with @yflim here: https://github.com/replikativ/persistent-sorted-set/tree/durability
